torch/csrc/autograd/engine.cpp

```diff
@@ -819,9 +819,16 @@ static variable_list call_tensor_pre_hooks(Node& fn, variable_list inputs) {
 static variable_list call_post_hooks(
     Node& fn,
     variable_list outputs,
-    const variable_list& inputs) {
+    const variable_list& inputs,
+    const bool had_post_hooks) {
   for (const auto& hook : fn.post_hooks()) {
-    outputs = (*hook)(outputs, inputs);
+    if (had_post_hooks) {
+      outputs = (*hook)(outputs, inputs);
+    } else {
+      variable_list null_inputs;
+      null_inputs.resize(inputs.size());
```

Comment:
You're not allowed to touch "inputs" after it's been moved!
You should either make this size 0 and grab the inputs size before the vector is moved.

Reply:
oh right